### ISSUE TITLE
exception when changing config which loads mesh

### ISSUE THREAD

**Issue by Karsten1987** [ros2/rviz#355]
The default rviz config has a subscription to a marker publisher, which loads a mesh.
When changing the config to another which also has the same subscription, the exception below is thrown and Rviz2 terminates.

```
[ERROR] [rviz2]: ItemIdentityException: Resource with the name file:///Users/karsten/workspace/osrf/roscon2018_ws/src/roscon2018/confbot_description/meshes/Shark_t.stl already exists. in ResourceManager::add at /Users/karsten/workspace/osrf/ros2_visualization/build/rviz_ogre_vendor/ogre-master-ca665a6-prefix/src/ogre-master-ca665a6/OgreMain/src/OgreResourceManager.cpp (line 158)
libc++abi.dylib: terminating with uncaught exception of type Ogre::ItemIdentityException: ItemIdentityException: Resource with the name file:///Users/karsten/workspace/osrf/roscon2018_ws/src/roscon2018/confbot_description/meshes/Shark_t.stl already exists. in ResourceManager::add at /Users/karsten/workspace/osrf/ros2_visualization/build/rviz_ogre_vendor/ogre-master-ca665a6-prefix/src/ogre-master-ca665a6/OgreMain/src/OgreResourceManager.cpp (line 158)
Abort trap: 6
```

**Comment by wjwwood** [comment]
Probably needs to either detect when the resource already exists and reuse it, or remove it first, or give each a unique resource name.

**Comment by routiful** [comment]
@Karsten1987  
I've got same error. Did you solve it?

**Comment by Karsten1987** [comment]
I believe this is still an open bug. As a workaround, you can edit your configuration manually or remove it altogether.

**Comment by mikaelarguedas** [comment]
I'm facing this using the dashing debs as well,
In my case I am not changing the config but simply disabling and reenabling the plugin using the resource.

**Comment by Martin-Idel** [comment]
@mikaelarguedas and @Karsten1987  I think I fixed the problem in #410 - if you have a source build, maybe you can validate?

We have a check whether a mesh already exists and in that case, it shouldn't be loaded twice, but there is a bug in the STL loader, loading the mesh into a different resource group so that the existence check doesn't apply.
